common/src/main/java/de/michelinside/glucodatahandler/common/receiver/NotificationReceiver.kt

```diff
@@ -0,0 +1,125 @@
+
+                    // Try processing different RemoteViews
+                    if (processRemoteViews(contentView)) return
+                    if (processRemoteViews(bigContentView)) return
+
+                    // If we get here, we couldn't find the glucose value
+                    Log.d(LOG_ID, "Could not find glucose value in notification")
+                }
+            }
+        }
+    }
+
+
+    private fun processRemoteViews(remoteViews: RemoteViews?): Boolean {
+        if (remoteViews == null) return false
+
+        try {
+            // Apply the RemoteViews to get the actual View hierarchy
+            val context = applicationContext
+            val applied = remoteViews.apply(context, null)
+            val root = applied.rootView as ViewGroup
+
+            // Collect all TextViews
+            val textViews = ArrayList<TextView>()
+            findTextViews(root, textViews)
+
+            Log.d(LOG_ID, "Found ${textViews.size} text views")
+
+            var matches = 0
+            var glucoseValue = 0
+
+            // Examine each TextView
+            for (textView in textViews) {
+                try {
+                    val bg = textView.text?.toString()?.toIntOrNull() ?: 0
+                    if (bg > 0) {
+                        glucoseValue = bg
+                        matches++
+                    }
+                } catch (e: Exception) {
+                    Log.e(LOG_ID, "Error processing TextView: ${e.message}")
+                }
+            }
+
+            textViews.clear()
+
+            when {
+                matches == 0 -> {
+                    Log.d(LOG_ID, "Did not find any matches")
+                    return false
+                }
+                matches > 1 -> {
+                    Log.e(LOG_ID, "Found too many matches: $matches")
+                    return false
+                }
+                else -> {
+                    handleGlucoseValue(glucoseValue)
+                    return true
+                }
+            }
+
+        } catch (e: Exception) {
+            Log.e(LOG_ID, "Error processing RemoteViews: ${e.message}")
+            return false
+        }
+    }
+
+    private fun findTextViews(view: View, textViews: MutableList<TextView>) {
+        when (view) {
+            is TextView -> textViews.add(view)
+            is ViewGroup -> {
+                for (i in 0 until view.childCount) {
+                    findTextViews(view.getChildAt(i), textViews)
+                }
+            }
+        }
+    }
+
+    private fun handleGlucoseValue(glucoseValue: Int) {
+        Log.d(LOG_ID, "Extracted glucose value: $glucoseValue")
+        // TODO: g7 notifications update weirdly often. deduplication is needed
+        if (GlucoDataUtils.isGlucoseValid(glucoseValue)) {
+            val glucoExtras = Bundle()
+            glucoExtras.putLong(ReceiveData.TIME, System.currentTimeMillis())
+            glucoExtras.putInt(ReceiveData.MGDL,glucoseValue)
```

Comment:
glucoseValue may also be a mmol/ value, so you should check using GlucoDataUtils.isMmolValue().
If it is you should also set the GLUCOSECUSTOM like here:
https://github.com/pachi81/GlucoDataHandler/blob/679aa3241dc39cc457148600c2cd71e59494ec57/common/src/main/java/de/michelinside/glucodatahandler/common/tasks/NightscoutSourceTask.kt#L192

Reply:
I added you as contributor, so it may be easier to create PRs.
https://github.com/pachi81/GlucoDataHandler/invitations